src/runtime/typemanager.cs

```diff
@@ -76,11 +77,22 @@ internal static void RestoreRuntimeData(RuntimeDataStorage storage)
         {
             Debug.Assert(cache == null || cache.Count == 0);
             storage.GetValue("slots", out _slotsImpls);
-            storage.GetValue("cache", out cache);
-            foreach (var entry in cache)
+            var _cache = new Dictionary<MaybeType, IntPtr>();
+            storage.GetValue("cache", out _cache);
```

Comment:
the value you put into `_cache` here will always be overwritten.

You can (and should) use `out var _cache`.